src/scheduler/buckets.c

```diff
@@ -1121,7 +1131,8 @@ check_node_buckets(status *policy, server_info *sinfo, queue_info *qinfo, resour
 				schdlog(PBSEVENT_DEBUG3, PBS_EVENTCLASS_JOB, LOG_DEBUG, resresv->name,  "Request won't fit into any placement sets, will use all nodes");
 				return map_buckets(policy, sinfo->buckets, resresv, err);
 			}
-		}
+		} else
+			move_schd_error(err, failerr);
```

Comment:
I'm not clear on the purpose of 'can_run', so I apologize for my ignorance, but if the job can run, won't we just return from the function on line 1117 ?

Reply:
regardless, can you please add a comment here to explain why you are moving failerr to err if can_run == 1 ?

Reply:
This one might deserve a comment.  The idea is if we can possibly run, we want to maintain that comment.  We don't want to return a comment for a placement set that the job can never run in.  That is the reason we were seeing never run messages.  There was a placement set that the job could run in, but just not right then.  We moved on and the final placement set we checked would never be possible.  We returned the comment for the final placement set.